ctest-next/templates/test.rs

```diff
@@ -0,0 +1,74 @@
+use std::mem;
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+
+{% if rustc_version < RustcVersion::new(1, 30, 0) %}
+    static FAILED: AtomicBool = std::sync::atomic::ATOMIC_BOOL_INIT;
+    static NTESTS: AtomicUsize = std::sync::atomic::ATOMIC_USIZE_INIT;
+{% else %}
+    static FAILED: AtomicBool = AtomicBool::new(false);
+    static NTESTS: AtomicUsize = AtomicUsize::new(0);
+{% endif %}
+
+fn main() {
+    eprintln!("RUNNING ALL TESTS");
+    run_all();
+    if FAILED.load(Ordering::SeqCst) {
+        panic!("some tests failed");
+    } else {
+        eprintln!("PASSED {} tests", NTESTS.load(Ordering::SeqCst));
+    }
+}
+
+trait Pretty {
+    fn pretty(&self) -> String;
+}
+
+impl<'a> Pretty for &'a str {
+    fn pretty(&self) -> String { format!("{:?}", self) }
+}
+impl<T> Pretty for *const T {
+    fn pretty(&self) -> String { format!("{:?}", self) }
+}
+impl<T> Pretty for *mut T {
+    fn pretty(&self) -> String { format!("{:?}", self) }
+}
+macro_rules! p {
+    ($($i:ident)*) => ($(
+        impl Pretty for $i {
+            fn pretty(&self) -> String {
+                format!("{} ({:#x})", self, self)
+            }
+        }
+    )*)
+}
+p! { i8 i16 i32 i64 u8 u16 u32 u64 usize isize }
+
+fn same<T: Eq + Pretty>(rust: T, c: T, attr: &str) {
```

Comment:
This should be something like `ensure_same` or `check_same` just to hint that it updates the test counters. Also we can get by with PartialEq rather than Eq right?